### ISSUE TITLE
Better naming of XML file

### ISSUE THREAD

**Issue by magnolia61** [Project60/sepa_dd#229]
Using different creditors and different groups, it would be nice if the filename resembles something of the creditor tag. I believe Olaf implemented something like this in:
https://github.com/sfe-ev/sepa_dd/commit/69852f7cd1a5f36cf5831548107a7378bb89864c
So
SDDXML-TXG-3-FRST-2015-02-01.xml
could be
SDDXML-TAG1-TXG-3-FRST-2015-02-01.xml
And all would be identifyable and sortable easily :-)

Maybe it is even better (cosmeticaly) to use the mandate prefix because it is always 4 characters

**Comment by Detsieber** [comment]
> I believe Olaf implemented something like this in:
> sfe-ev@
> <https://github.com/sfe-ev/sepa_dd/commit/69852f7cd1a5f36cf5831548107a7378bb89864c>
yes, and meanwhile he also implemented proper naming for COR1-files
(national "quick" SEPA DD with only 1 advance day), and CORE files for
international SEPA DD, such as:

SDDXML_DICO_20150107_COR1_55.xml
or
SDDXML_DICO_20150112_CORE_57.xml

(where "DICO" is our mandate prefix).




Am 13.01.2015 um 00:22 schrieb Richard van Oosterhout:

**Comment by systopia** [comment]
I can see why this would be helpful, but using the tags might not be the way. Tags can change. @Detsieber's mandate prefix might be a good option. But I'm open to more suggestions.

**Comment by antrik** [comment]
No, the Creditor Tag is really the right thing to use here. It was introduced for the explicit purpose of allowing transactions for Creditors sharing a single bank account to go into the same XML file; so the Tag uniquely identifies the Creditor group for which the file is generated -- while Mandate Prefixes could differ within the group, and/or be identical among Creditors in different groups.

(In the SfE variant at least support for Creditor Tags is not fully implemented though, as we don't have an actual use case for it so far... Don't know about the systopia/upstream variant.)

@magnolia61: there is no requirement for the Mandate Prefix to always be exactly four characters. For some reason it has been arbitrarily limited to a *maximum* of four characters -- see issue #85. (Not sure why this one is closed -- as far as I can see, the limitation still exists in the systopia/upstream variant as well...)

**Comment by antrik** [comment]
Oh, one more bit I forgot: I don't see how changing Tags would be a problem. The Tag in the file name states for which Creditor group the file was generated -- if the Tags/grouping change later, that doesn't (or at least shouldn't) affect previously generated files, while newly generated files should clearly carry the new Tag.

**Comment by systopia** [comment]
With #314 it will become easier to implement specific naming schemes per format.
